requirements.txt

```diff
@@ -1,23 +1,34 @@
 # We're waiting on Django 3 until it's further into the lifecycle.
-Django>=2.2.9,<2.2.99
-djangosaml2==0.17.2
+Django>=3.2.9,<3.2.99
+djangosaml2==1.3.6
+
+# Hasn't been updated since 2018
 django-cron>=0.5.1,<0.5.99
-django-debug-toolbar>=2.1,<2.1.99
-django-extensions>=2.2.6,<2.2.99
-django-mysql>=3.3.0,<3.3.99
+django-debug-toolbar>=3.2,<3.2.99
+django-extensions>=3.1.5,<3.1.99
+django-mysql>=4.5.0,<4.5.99
+
+# Hasn't been updated since 2016
 django-npm==1.0.0
+
+# This needs to be switched to debugpy
 django-ptvsd-debug==1.0.3
-django-redis>=4.11.0,<4.11.99
-django-registration-redux==2.7
+django-registration-redux==2.9
+
+#This hasn't been updated since 2016
 django-settings-export>=1.2.1,<1.2.99
-django-watchman>=1.0.0,<1.0.99
-libsass>=0.19.4,<0.19.99
-mysqlclient>=1.4.6,<1.4.99
+django-watchman>=1.2.0,<1.2.99
+libsass>=0.21.0,<0.21.99
+mysqlclient>=2.1.0,<2.1.99
+
+#This needs to be switched to debugpy
 ptvsd>=4.3.2,<4.3.99
-python-dateutil>=2.8.1,<2.8.99
-python-decouple==3.3
-python-ldap==3.4.0
-requests>=2.22.0,<2.22.99
-whitenoise>=5.0.1,<5.0.9
-xlrd==1.2.0
-xlwt==1.3.0
\ No newline at end of file
+
+python-dateutil>=2.8.2,<2.8.99
+python-decouple==3.6
+requests>=2.27.0,<2.27.99
+whitenoise>=6.0.0,<6.0.99
+xlrd==2.0.1
+
+# This hasn't been updated since 2017
+xlwt==1.3.0
```

Comment:
@jonespm, looks like the bump to `2.0.1` broke the application's ability to handle `.xlsx` files. I dunno exactly what to do about this.

See the release notes about not supporting `.xlsx`: https://pypi.org/project/xlrd/
This appears to be an alternative: https://openpyxl.readthedocs.io/en/stable/

Reply:
I found this when uploading cohort data.

Reply:
Hmm, okay thanks for testing. It still will read **.xls** files though. I don't know how common uploading **.xlsx** is or if that's even a requirement. I guess we'll either have to upgrade or leave it at 1.2.0 then. 

I'll send an email to Hema.

Reply:
Seems somewhat disruptive if the default file type for Excel is `.xlsx` now. I did test with `.xls`, it worked.

Reply:
If it was more than Hema uploading this I'd agree, but since it's just one person and we plan to deprecate this app. It might not be worth the time to update this. I can create a new issue for it though.

Reply:
Hema said she just pastes into the member text box and never even uploads any Excel file. So this won't matter for her process. I'll create it as a new issue incase we did want to expand the usage or fix this.

Reply:
Okay, thanks for creating the issue.